### PR TITLE
Remove Module.buffer

### PR DESCRIPTION
**Pull request by kripken**
We don't need to put it on the Module object - that only has downsides in code size and clarity.

We do still look if the user provided Module.buffer as an input, as before, so this PR has no functional changes.

### CODE REVIEW

File: src/preamble.js [hunk 1]
```diff
@@ -322,7 +322,7 @@ var HEAP,
   HEAPF64;
 
 function updateGlobalBuffer(buf) {
```
Comment (sbc100) [inline src/preamble.js]: I think you can remove this function completely.

It has exactly one call site and the callsite which always used the return value of `emscripten_realloc_buffer()`, but `emscripten_realloc_buffer()` already takes care up updating the global `buffer`.

Comment (kripken) [reply src/preamble.js]: Thanks, done.

File: src/library.js [hunk 2]
```diff
@@ -504,14 +504,14 @@ LibraryManager.library = {
 #if WASM
     var PAGE_MULTIPLE = {{{ getPageSize() }}};
     size = alignUp(size, PAGE_MULTIPLE); // round up to wasm page size
-    var old = Module['buffer'];
+    var old = buffer;
     var oldSize = old.byteLength;
```
Comment (sbc100) [inline src/library.js]: no need for `old` here.. just `var oldSize = buffer.byteLength;`

Comment (sbc100) [reply src/library.js]: Also this function seems a little confused about its return type.  Perhaps it should simply return true on sucess and false on failure?   No need to do that as part of this CL though I guess.

Comment (kripken) [reply src/library.js]: Fixed the `old` stuff, thanks.

Let's think about the return type later, yeah. Offhand it looks like we check if we return a buffer, and its size - I don't remember why exactly, but might be some reason we can't just return a bool.

**Comment by VirtualTim** [comment]
Should this also be removed from threads? Eg here: https://github.com/emscripten-core/emscripten/blob/incoming/src/worker.js#L114

**Comment by kripken** [comment]
Good question - I'm not sure. I think that line is ok actually, but we should read from Module as well, in another place (that is, we have to write to Module since we are not in the right scope to write to the var; and then we should read from it as well). But the same is true for `wasmModule`, which was not changed in this PR - so either this was broken before, and not tested, or we are missing something.

@juj, what do you think?

**Comment by rajsite** [comment]
@kripken Apparently [some of our source was using Module.buffer](https://github.com/ni/VireoSDK/blob/b9f34b965811f6c1db6fdcfddde2190c5978d7f4/source/io/module_eggShell.js#L456). Cleaning it up isn't so bad but it might be nice to mention in the [Changelog for 1.38.30](https://github.com/emscripten-core/emscripten/blob/incoming/ChangeLog.md) that something was removed from Module.

**Comment by kripken** [comment]
Thanks @rajsite , good idea, I added that to the changelog now.

**Pull request merged**